common/src/main/java/com/wynntils/features/inventory/ContainerSearchFeature.java

```diff
@@ -212,14 +219,21 @@ private void addSearchWidget(AbstractContainerScreen<?> screen, int renderX, int
         screen.addRenderableWidget(lastSearchWidget);
     }
 
-    private void matchItems(String searchStr, AbstractContainerScreen<?> screen) {
+    private void matchItems(String searchStr, ChestMenu chestMenu) {
         String search = searchStr.toLowerCase(Locale.ROOT);
 
-        NonNullList<ItemStack> playerItems = McUtils.inventory().items;
-        for (ItemStack itemStack : screen.getMenu().getItems()) {
+        Container container = chestMenu.getContainer();
+        for (int i = 0; i < container.getContainerSize(); i++) {
+            if ((currentSearchableContainerType == SearchableContainerType.GUILD_BANK
```

Comment:
Somewhere along the line you lost the code that had the "content" part of the container encoded as part of each individual SearchableContainerType enum value. I think that was a good design idea. The implementation you had at that time was lacking somewhat (it was overly complex with a new type which was not needed) but the core of the idea was sound, I think, to let the enums know if an index was inside the "content" area or not.

Reply:
i don't think that was needed with only two values and I also should keep the change to a minimum but I hope we can someday change it when we have a better system to handle wynncrafts custom UIs